Source/Particles/PhysicalParticleContainer.cpp

```diff
@@ -1315,7 +1318,16 @@ PhysicalParticleContainer::SplitParticles(int lev)
     for (WarpXParIter pti(*this, lev); pti.isValid(); ++pti)
     {
         pti.GetPosition(xp, yp, zp);
+
+        // offset for split particles is computed as a function of cell size
+        // and number of particles per cell, so that a uniform distribution
+        // before splitting results in a uniform distribution after splitting
+        const amrex::Vector<int> ppc_nd = plasma_injector->num_particles_per_cell_each_dim;
         const std::array<Real,3>& dx = WarpX::CellSize(lev);
+        amrex::Vector<amrex::Real> split_offset = {dx[0]/2/ppc_nd[0],
```

Comment:
I would feel more secure if you wrote this as floating point operation :)
```suggestion
        amrex::Vector<amrex::Real> split_offset = {dx[0]/2./ppc_nd[0],
```
or even better
```suggestion
        amrex::Vector<amrex::Real> split_offset = {dx[0]/amrex::Real(2*ppc_nd[0]),
```

Reply:
Just because
```
// main.cpp
#include <iostream>

int main()
{
    double const r = 1.;
    double const V1 = 4 / 3 * 3.1415 * r * r * r;
    double const V2 = 4. / 3 * 3.1415 * r * r * r;
    std::cout << V1 << " " << V2 << std::endl;
}
```
```bash
$ g++ main.cpp && ./a.out 
3.1415 4.18867
```

Reply:
fixed in eee301aa9c25cc9efa859728df20a9e5bf47ae95